lib/screens/onboarding/continue_recovery.dart

```diff
@@ -0,0 +1,468 @@
+import 'dart:async';
+
+import 'package:eosdart_ecc/eosdart_ecc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:intl/intl.dart';
+import 'package:seeds/constants/app_colors.dart';
+import 'package:seeds/models/models.dart';
+import 'package:seeds/providers/notifiers/settings_notifier.dart';
+import 'package:seeds/providers/services/eos_service.dart';
+import 'package:seeds/providers/services/http_service.dart';
+import 'package:seeds/widgets/main_button.dart';
+import 'package:seeds/widgets/notion_loader.dart';
+import 'package:seeds/widgets/second_button.dart';
+import 'package:share/share.dart';
+import 'package:seeds/i18n/wallet.i18n.dart';
+
+enum RecoveryStatus {
+  loading,
+  waitingConfirmations,
+  waitingTimelock,
+  claimReady,
+  noGuardiansFound
+}
+
+class ContinueRecovery extends StatefulWidget {
+  final Function onClaimed;
+  final Function onBack;
+
+  ContinueRecovery(this.onClaimed, this.onBack);
+
+  @override
+  _ContinueRecoveryState createState() => _ContinueRecoveryState();
+}
+
+class _ContinueRecoveryState extends State<ContinueRecovery> {
+
+  bool recovering = false;
+  bool canClaim = false;
+  bool doneSuccess = false;
+
+  @override
+  Widget build(BuildContext context) {
+    String accountName = SettingsNotifier.of(context).accountName;
+
+    if (recovering) {
+      return NotionLoader(
+        notion: "Recovering account...",
+      );
+    } else if (doneSuccess) {
+      return successComponent(accountName);
+    } else if (canClaim) {
+      return claimReadyComponent(accountName);
+    }
+
+    return FutureBuilder(
+        future: findRecovery(accountName),
+        builder: (context, snapshot) {
+          var status = RecoveryStatus.loading;
+
+          UserRecoversModel recovers;
+          UserGuardiansModel guardians;
+
+          int confirmedGuardians = 0;
+          int requiredGuardians;
+          int timeLockSeconds = 0;
+
+          if (snapshot.connectionState == ConnectionState.done &&
+              snapshot.hasData) {
+            recovers = snapshot.data[0];
+            guardians = snapshot.data[1];
+
+            print("recovers: ${recovers.exists}");
+            print("guardians: ${guardians.exists}");
+
+            if (!guardians.exists) {
+              status = RecoveryStatus.noGuardiansFound;
+            } else {
+              status = RecoveryStatus.waitingConfirmations;
+              requiredGuardians = guardians.guardians.length;
+              if (!recovers.exists) {
+                confirmedGuardians = 0;
+              } else {
+
+                confirmedGuardians = recovers.guardians.length;
+                timeLockSeconds = recovers.completeTimestamp + guardians.timeDelaySec;
+
+                if ((requiredGuardians == 3 && confirmedGuardians >= 2) ||
```

Comment:
This is logic that's also in the contract. 

Not easy to extract from this since we need basically all intermediary results in the widget, to display the exact details, who signed, who didn't sign yet, and so on. 

I tried to make it a function but didn't make sense. 

Will add some comments

Reply:
Renamed variables to make more sense, added comments.

Reply:
I mean - it could be a view model object that contains all the info we want to show

- people who signed
- how many still have to sign
- when all of them signed, what's the time lock date